sdk/nodejs/tests/runtime/props.spec.ts

```diff
@@ -161,12 +161,32 @@ describe("runtime", () => {
                                         ...isSecret && { secret: isSecret },
                                         ...(deps.length > 0) && { dependencies: deps },
                                     },
+                                    expectedRoundTrip: new Output(resources,
+                                        Promise.resolve(isKnown ? tv.expected : undefined), Promise.resolve(isKnown),
+                                        Promise.resolve(isSecret), Promise.resolve([])),
```

Comment:
The last element is weird, I am guessing this corresponds to
```
    /** @internal */
    public constructor(
        resources: Set<Resource> | Resource[] | Resource,
        promise: Promise<T>,
        isKnown: Promise<boolean>,
        isSecret: Promise<boolean>,
        allResources: Promise<Set<Resource> | Resource[] | Resource> | undefined) {
```

Is it possible to skip it (default undefined)? 

It's weird that the constructor has two parameters where resources can be passed in but there are probably reasons.

Reply:
> It's weird that the constructor has two parameters where resources can be passed in but there are probably reasons.

Yeah, it's a little strange. It was introduced in #3680. I'm not entirely familiar with the backstory, but looks like it was done this way to address some SxS problems.

> Is it possible to skip it (default undefined)?

We might be able to make it optional with a default value of undefined.